src/vkscript/command_parser.cc

```diff
@@ -627,15 +627,15 @@ Result CommandParser::ProcessSSBO() {
         ((cmd->GetOffset() / buf->GetFormat()->SizeInBytes()) *
          buf->GetFormat()->InputNeededPerElement()) +
         static_cast<uint32_t>(values.size());
-    // The buffer should only be resized to become bigger. This means that if a
-    // command was run to set the buffer size we'll honour that size until a
-    // request happens to make the buffer bigger.
-    if (value_count > buf->ValueCount())
-      buf->SetValueCount(value_count);
-
-    // Even if the value count doesn't change, the buffer is still resized
-    // because this maybe the first time data is set into the buffer.
-    buf->SetSizeInBytes(buf->GetSizeInBytes());
+    uint32_t element_count = value_count;
+    if (buf->GetFormat()->GetPackSize() == 0) {
+      // This divides by the needed input values, not the values per element.
+      // The assumption being the values coming in are read from the input,
+      // where components are specified. The needed values maybe less then the
+      // values per element.
+      element_count = value_count / buf->GetFormat()->InputNeededPerElement();
+    }
+    buf->SetMaxSizeInBytes(element_count * buf->GetFormat()->SizeInBytes());
```

Comment:
This new implementation allows the max-size-in-bytes to shrink.
I think that can happen if you have an ssbo subdata command with a big offset, then a later one with a small offset. That's because value_count computed a few lines above takes into account the offset of *this* command but not the prior size.

Reply:
You're right I forgot about this condition:
    if(buf->GetMaxSizeInBytes() > element_count * buf->GetFormat()->SizeInBytes())
      buf->SetMaxSizeInBytes(element_count * buf->GetFormat()->SizeInBytes());
Does this fix the issue you've mentioned?